src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,222 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Instantiates and performs failure detection and handling asynchronously.
+ * <p>
+ * Failure Detector:
+ * Executes detection policy (eg. PeriodicPollingPolicy).
+ * It then checks for status of the nodes. If the result map is not empty,
+ * there are failed nodes to be addressed. This then triggers the failure
```

Comment:
Result map is just the result of the statuses of the servers received after polling.
You are right, the management server is a lot more than just failure detection.
Since I was explaining the failure detector, I have mentioned failure detector on top.

The failure detector of the management server is supposed to periodically execute the policy and depending on the status trigger the failure handler.